src/js/components/sharedComponents/autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,242 @@
+/**
+ * Created by michaelbray on 1/27/17.
+ */
+
+import React from 'react';
+import _ from 'lodash';
+
+import Warning from './Warning';
+import SuggestionHolder from './SuggestionHolder';
+
+const propTypes = {
+    values: React.PropTypes.array,
+    placeholder: React.PropTypes.string,
+    handleTextInput: React.PropTypes.func,
+    onSelect: React.PropTypes.func,
+    errorHeader: React.PropTypes.string,
+    errorMessage: React.PropTypes.string,
+    tabIndex: React.PropTypes.number,
+    isRequired: React.PropTypes.bool
+};
+
+const defaultProps = {
+    values: [],
+    placeholder: '',
+    tabIndex: null,
+    isRequired: false,
+    errorHeader: null,
+    errorDescription: null
+};
+
+export default class Autocomplete extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.dataDictionary = {};
+
+        this.state = {
+            value: '',
+            shown: '',
+            selectedIndex: 0,
+            showWarning: ''
+        };
+    }
+
+    componentDidMount() {
+        this.setupAutocomplete();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!_.isEqual(prevProps.values, this.props.values)) {
+            this.open();
+        }
+    }
+
+    onChange(e) {
+        this.checkValidity(e.target.value);
+        this.props.handleTextInput(e);
+    }
+
+    setupAutocomplete() {
+        const target = this.autocompleteInput;
+
+        target.addEventListener('blur', () => {
+            this.close();
+            target.value = "";
+        });
+
+        // enable tab keyboard shortcut for selection
+        target.addEventListener('keydown', (e) => {
+            // Enter
+            if (e.keyCode === 13) {
+                e.preventDefault();
+                this.select(this.props.values[this.state.selectedIndex]);
+                target.value = "";
+            }
+            // Tab or Escape
+            else if (e.keyCode === 9 || e.keyCode === 27) {
+                target.value = "";
+                this.close();
+            }
+            // Previous
+            else if (e.keyCode === 38) {
+                this.previous();
+            }
+            // Next
+            else if (e.keyCode === 40) {
+                this.next();
+            }
+        });
+    }
+
+    close() {
+        this.setState({
+            shown: 'hidden'
+        });
+    }
+
+    open() {
+        this.setState({
+            shown: ''
+        });
+    }
+
+    previous() {
+        if (this.state.selectedIndex > 0) {
+            this.setState({
+                selectedIndex: this.state.selectedIndex -= 1
+            });
+        }
+    }
+
+    next() {
+        if (this.state.selectedIndex < 9) {
```

Comment:
If you make the suggestion list length a prop (see comment in `SuggestionHolder`), you'll also want to modify this number.